### ISSUE TITLE
Increase robustness of ROOTSYS detection foo (Trac #587)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#586]
still often get strangish errors from ROOT voodoo (rootcint sometimes)

Often attached to having non-ports ROOTSYS in path or LD_LIB path or
env.

See, for example, email from Seon-hee:

Hi all again,

When I build before, I was setting 
ROOTSYS=/data1/i3_ports-v3/root-v5.20.00 .
When I do not set this as Henrik suggested,
the compilation goes fine.

Thanks!
(and sorry for the spam,)
Seon-Hee

Seon-Hee Seo wrote:
Hi all again,

I just needed to include dependent projects, simclasses and trigger-sim.
Now it works for cmake part.
(Thanks Henrik and RobertF.)

But my compile fails with the following error.

-------------------------------------
[ 56%] Generating 
/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.cxx 
with rootcint
/tmp/gab3A1_cint.cxx:1:13: error: C: No such file or directory
Error: external preprocessing failed. /usr/include/c++/4.1.3/exception:42:
Error: Too many '}' /usr/include/c++/4.1.3/exception:103:
/tmp/FsefFB_cint.cxx:1:13: error: C: No such file or directory
Error: external preprocessing failed. /usr/include/c++/4.1.3/typeinfo:42:
Error: Symbol __constchar is not defined in current scope  
/usr/include/gconv.h:118:
Error: Symbol __modname is not defined in current scope  
/usr/include/gconv.h:118:
Error: class,struct,union or type __builtin_va_list not defined  
/usr/lib/gcc/x86_64-linux-gnu/4.1.3/include/stdarg.h:43:
Error: Too many '}' /usr/include/stdio.h:841:
Error: Too many '}' /usr/include/nl_types.h:53:
Syntax error /usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:
Error: unrecognized language construct 
/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:
Error: Symbol __uselocale is not defined in current scope  
/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:
Error: Symbol __constchar is not defined in current scope  
/usr/include/time.h:148:
Error: Symbol tm_zone is not defined in current scope  
/usr/include/time.h:148:
Error: Too many '}' /usr/include/pthread.h:1104:
Error: Too many '}' /usr/include/unistd.h:1069:
Syntax Error: void* 
/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/gthr-default.h:113:
Error: Too many '}' /usr/include/ctype.h:218:
Error: Too many '}' /usr/include/c++/4.1.3/bits/functexcept.h:91:
Error: class,struct,union or type ... not defined  
/usr/include/c++/4.1.3/bits/cpp_type_traits.h:83:
Internal error: global function template arg type 
/usr/include/c++/4.1.3/bits/cpp_type_traits.h:83:
Error: Missing closing brace for the block opened around line 44.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 30.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 28.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 217.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 31.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 30.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Error: Missing closing brace for the block opened around line 42.
Error: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:
Warning: Error occurred during reading source files
Warning: Error occurred during dictionary source generation
!!!Removing 
/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.cxx 
/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.h !!!
Error: rootcint: error loading headers...
make[2]: *** [simclasses/CMakeFiles/simclassesDict.cxx] Error 1
make[1]: *** [simclasses/CMakeFiles/simclasses.dir/all] Error 2
make: *** [all] Error 2
---------------------------------

Could this be ROOT version issue?
How to deal with this?

Cheers,
Seon-Hee

Seon-Hee Seo wrote:

Hi all,

I was able to build V03-01-00 icerec.
But when I include sim-services (most recent release V00-07-01),
I get cmake error:

--------------------------------------

....
--   sim-services
CMake Error at cmake/project.cmake:82 (message):
 Attempt to use nonexistent project 'simclasses'
Call Stack (most recent call first):
 cmake/project.cmake:208 (use_projects)
 sim-services/CMakeLists.txt:6 (i3_add_library)

-- Configuring incomplete, errors occurred!
make: *** [rebuild_cache] Error 1
----------------------------------------

In V2 icerec, I did not have this problem.

I need sim-services project to change RunID, EventID
to be able to use data which contains same RunID, EventID pair.
(So that my analysis-tree handles the data properly.)

This data set is corsika set 2267 being used for filter study.

Any suggestion?

Thanks,
Seon-Hee
_______________________________________________

Migrated from https://code.icecube.wisc.edu/ticket/587
```json
{
    "status": "closed", 
    "changetime": "2013-06-28T13:01:06", 
    "description": "still often get strangish errors from ROOT voodoo (rootcint sometimes)\n\nOften attached to having non-ports ROOTSYS in path or LD_LIB path or\nenv.\n\nSee, for example, email from Seon-hee:\n\nHi all again,\n\nWhen I build before, I was setting \nROOTSYS=/data1/i3_ports-v3/root-v5.20.00 .\nWhen I do not set this as Henrik suggested,\nthe compilation goes fine.\n\nThanks!\n(and sorry for the spam,)\nSeon-Hee\n\nSeon-Hee Seo wrote:\nHi all again,\n\nI just needed to include dependent projects, simclasses and trigger-sim.\nNow it works for cmake part.\n(Thanks Henrik and RobertF.)\n\nBut my compile fails with the following error.\n\n-------------------------------------\n[ 56%] Generating \n/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.cxx \nwith rootcint\n/tmp/gab3A1_cint.cxx:1:13: error: C: No such file or directory\nError: external preprocessing failed. /usr/include/c++/4.1.3/exception:42:\nError: Too many '}' /usr/include/c++/4.1.3/exception:103:\n/tmp/FsefFB_cint.cxx:1:13: error: C: No such file or directory\nError: external preprocessing failed. /usr/include/c++/4.1.3/typeinfo:42:\nError: Symbol __constchar is not defined in current scope  \n/usr/include/gconv.h:118:\nError: Symbol __modname is not defined in current scope  \n/usr/include/gconv.h:118:\nError: class,struct,union or type __builtin_va_list not defined  \n/usr/lib/gcc/x86_64-linux-gnu/4.1.3/include/stdarg.h:43:\nError: Too many '}' /usr/include/stdio.h:841:\nError: Too many '}' /usr/include/nl_types.h:53:\nSyntax error /usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:\nError: unrecognized language construct \n/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:\nError: Symbol __uselocale is not defined in current scope  \n/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/c++locale.h:55:\nError: Symbol __constchar is not defined in current scope  \n/usr/include/time.h:148:\nError: Symbol tm_zone is not defined in current scope  \n/usr/include/time.h:148:\nError: Too many '}' /usr/include/pthread.h:1104:\nError: Too many '}' /usr/include/unistd.h:1069:\nSyntax Error: void* \n/usr/include/c++/4.1.3/x86_64-linux-gnu/bits/gthr-default.h:113:\nError: Too many '}' /usr/include/ctype.h:218:\nError: Too many '}' /usr/include/c++/4.1.3/bits/functexcept.h:91:\nError: class,struct,union or type ... not defined  \n/usr/include/c++/4.1.3/bits/cpp_type_traits.h:83:\nInternal error: global function template arg type \n/usr/include/c++/4.1.3/bits/cpp_type_traits.h:83:\nError: Missing closing brace for the block opened around line 44.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 30.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 28.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 217.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 31.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 30.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nError: Missing closing brace for the block opened around line 42.\nError: Unexpected end of file (G__exec_statement()) /tmp/Bd3P2a_cint.cxx:2:\nWarning: Error occurred during reading source files\nWarning: Error occurred during dictionary source generation\n!!!Removing \n/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.cxx \n/home/seo/icerec/V3-1-0/build/simclasses/CMakeFiles/simclassesDict.h !!!\nError: rootcint: error loading headers...\nmake[2]: *** [simclasses/CMakeFiles/simclassesDict.cxx] Error 1\nmake[1]: *** [simclasses/CMakeFiles/simclasses.dir/all] Error 2\nmake: *** [all] Error 2\n---------------------------------\n\nCould this be ROOT version issue?\nHow to deal with this?\n\nCheers,\nSeon-Hee\n\nSeon-Hee Seo wrote:\n\nHi all,\n\nI was able to build V03-01-00 icerec.\nBut when I include sim-services (most recent release V00-07-01),\nI get cmake error:\n\n--------------------------------------\n\n....\n--   sim-services\nCMake Error at cmake/project.cmake:82 (message):\n Attempt to use nonexistent project 'simclasses'\nCall Stack (most recent call first):\n cmake/project.cmake:208 (use_projects)\n sim-services/CMakeLists.txt:6 (i3_add_library)\n\n\n-- Configuring incomplete, errors occurred!\nmake: *** [rebuild_cache] Error 1\n----------------------------------------\n\nIn V2 icerec, I did not have this problem.\n\nI need sim-services project to change RunID, EventID\nto be able to use data which contains same RunID, EventID pair.\n(So that my analysis-tree handles the data properly.)\n\nThis data set is corsika set 2267 being used for filter study.\n\nAny suggestion?\n\nThanks,\nSeon-Hee\n_______________________________________________", 
    "reporter": "blaufuss", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1372424466000000", 
    "component": "cmake", 
    "summary": "Increase robustness of ROOTSYS detection foo", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2010-01-19T20:53:52", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20100120T00:36:07: tfs changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20100420T19:53:27: 
* nega changed status from "assigned" to "new"
* nega changed owner from "tfs" to "nega"

**Comment by icecube-trac** [comment]
Trac update at 20100505T15:46:56: nega changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20130628T13:01:06: 
* nwhitehorn changed status from "assigned" to "closed"
* nwhitehorn changed resolution from "" to "fixed"
* nwhitehorn commented:

> This seems to have been fixed in r1612 (ticket #364)
